app/backend/approaches/chatreadretrievereadvision.py

```diff
@@ -123,9 +125,9 @@ async def run_until_final_call(
         if use_vector_search:
             for field in vector_fields:
                 vector = (
-                    await self.compute_text_embedding(query_text)
-                    if field == "embedding"
-                    else await self.compute_image_embedding(query_text)
+                    await self.compute_image_embedding(query_text)
+                    if field.startswith("image")
```

Comment:
This was a bit tricky, feels a bit code smelly.

Reply:
maybe if field is self.embedding field?